src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Utilities/UIThreadHelper.cs

```diff
@@ -11,26 +11,27 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Utilities
     internal static class UIThreadHelper
     {
         /// <summary>
-        /// Helper utility to ensure that we are on UI thread. Needs to be called 
-        /// in every method/propetrty needed UI thread for our protection (to avoid hangs 
+        /// Helper utility to ensure that we are on UI thread. Needs to be called
+        /// in every method/property with UI thread affinity (to avoid hangs
         /// which are hard to repro and investigate).
         /// </summary>
         public static void VerifyOnUIThread([CallerMemberName] string memberName = "")
```

Comment:
If `VerifyOnUIThread` is only used in Debug configuration, but it still used in `GetUnconfiguredProject()` and `GetDTEProject()`, should `UIThreadHelper` be renamed to something more specific like DebugOnlyUIThreadHelper ?

Reply:
> If VerifyOnUIThread is only used in Debug configuration

It's used in all configurations. It just doesn't assert in release builds. The `#if` here is just a minor perf thing that avoids the overhead of having `try`/`catch` in the method for release builds.